source/common/protobuf/utility.cc

```diff
@@ -12,10 +13,24 @@ MissingFieldException::MissingFieldException(const std::string& field_name,
           fmt::format("Field '{}' is missing in: {}", field_name, message.DebugString())) {}
 
 void MessageUtil::loadFromFile(const std::string& path, Protobuf::Message& message) {
-  const std::string json = Filesystem::fileReadToEnd(path);
-  const auto status = Protobuf::util::JsonStringToMessage(json, &message);
+  const std::string file_contents = Filesystem::fileReadToEnd(path);
+  // If the filename ends with .pb, do a best-effort attempt to parse it as a proto.
+  if (path.size() > 3 && path.substr(path.size() - 3) == ".pb") {
```

Comment:
I'm actually leaning now to not supporting ASCII PB at all. After chatting with @louiscryan and digging into this, it seems it doesn't have consistent cross-language support and JSON is considered the official proto3 textual representation. In addition, the opaque filter config we have today look horrible in ASCII PB, so we should basically just stick to JSON (see https://gist.github.com/htuch/aa273c9bf01ba9c6b18353ec8799695e for an example).